### PR TITLE
[SYCL] Add ITT annotations for SPIR functions as well

### PR DESCRIPTION
**Pull request by MrSidims**
Previously if was done one for SPIR kernels

Signed-off-by: Dmitry Sidorov <dmitry.sidorov@intel.com>

### CODE REVIEW

**Comment by MrSidims** [comment]
- [ ] Add a test

**Review by vzakhari — DISMISSED** [review]
Thanks!

File: llvm/lib/Transforms/Instrumentation/SPIRITTAnnotations.cpp [hunk 1]
```diff
@@ -240,20 +246,24 @@ PreservedAnalyses SPIRITTAnnotationsPass::run(Module &M,
       SPIRV_GROUP_FMAX,      SPIRV_GROUP_UMAX, SPIRV_GROUP_SMAX};
 
   for (Function &F : M) {
-    // Annotate only SPIR kernels
-    if (F.isDeclaration() || !isSPIRKernel(F))
+    // Annotate only SPIR kernels and functions
+    bool IsSPIRKernel = isSPIRKernel(F);
+    bool IsSPIRFunction = isSPIRFunction(F);
+    if (F.isDeclaration() || !(isSPIRKernel || isSPIRFunction))
```
Comment (bader) [inline llvm/lib/Transforms/Instrumentation/SPIRITTAnnotations.cpp:252]:
```suggestion
    if (F.isDeclaration())
```
`(isSPIRKernel || isSPIRFunction)` is `true` for the module with SPIR target triple.

File: llvm/lib/Transforms/Instrumentation/SPIRITTAnnotations.cpp [hunk 2]
```diff
@@ -138,11 +138,17 @@ ModulePass *llvm::createSPIRITTAnnotationsPass() {
...
+inline bool isSPIRKernel(Function &F) {
   return F.getCallingConv() == CallingConv::SPIR_KERNEL;
 }
 
+// Check for calling convention of a function. Return true if it's SPIR
+// function.
+inline bool isSPIRFunction(Function &F) {
+  return F.getCallingConv() == CallingConv::SPIR_FUNC;
+}
+
```
Comment (bader) [inline llvm/lib/Transforms/Instrumentation/SPIRITTAnnotations.cpp:151]: This code is not needed.

File: llvm/lib/Transforms/Instrumentation/SPIRITTAnnotations.cpp (hunk 1 above)
Comment (MrSidims) [reply llvm/lib/Transforms/Instrumentation/SPIRITTAnnotations.cpp:252]: Applied

**Review by bader — DISMISSED** [review]

**Review by bader — APPROVED** [review]

**Comment by bader** [comment]
@MrSidims, please, take a look at pre-commit failures.

**Comment by MrSidims** [comment]
> @MrSidims, please, take a look at pre-commit failures.

Taking a look

**Comment by MrSidims** [comment]
@bader I have restarted precommit to see if its sporadic. TBH I have no clue, how this patch could affect the failure: under gdb compilation of the test never passes through "SPIRITTAnnotationsPass::run" function. Also local execution of the test results in an assertion `(BT->getBitWidth() == 64 || (ConstValue >> BT->getBitWidth()) == 0) && "Size of externally provided specialization constant value doesn't" "fit into the specialization constant type"' with a compiler build of 20th of March.

**Pull request merged**